source-academy/frontend
# Add agenda and stash visualisation to environment visualiser

### Description

This pull request updates the environment visualiser by adding agenda and stash visualisation as part of the steps to revamp the environment visualiser.

### Type of change


### How to test

1. Run your local version of Source Academy linked to your own local js-slang. Make sure your local js-slang version is from https://github.com/source-academy/js-slang/pull/1430.
2. Enter a program in the playground, and switch to the environment visualizer tab.
3. Run the program and step through to see the changes.
4. The program can be edited and then rerun to see the updated environment.

### Checklist


### Future Work
- Continue to improve and revamp the overall look and feel of the environment visualiser.
- Refactor code for agenda and stash. See https://github.com/source-academy/frontend/commit/08787e29218553fb4be86eb2a89841aa5e3522ac.
- Line highlighting for the editor also probably needs a refactor. It is currently being done via `public/externalLibs/inspector/inspector.js`. The reducer used for the highlight action is currently redundant. See https://github.com/source-academy/frontend/commit/9bfca78fac579d622188bdfb8d5062d2bbb809d5 and https://github.com/source-academy/frontend/commit/ced59d455538c84bf7e60d415ace23074672e769.
- Add testing

### Note
- This pull request needs the update in js-slang and would not work without https://github.com/source-academy/js-slang/pull/1430.
- I needed to remove "obsolete" snapshot files for `ApplicationWrapper` and `ResearchAgreementPrompt` if not the test suite would complain. I am not sure if this is correct as I was resolving merge conflicts when merging with main.
